### DISCUSSION TITLE
Save problem with Thonny

### DISCUSSION THREAD

**Discussion by rapzak** [micropython/micropython#10586] [General]
Hi,
I hope this is not a silly question and has landed the right spot :)

I have a problem in thonny, tha randomly, maybe every 10 saves, i get this error while saving the PY file.

I am running micropython on a custom board, with an external ttl to usb adapter.
I am running on STM32WB55.
In general things works stable...

When i press the board in lower right corner it re-connects and work again - no need to power cycle or anything

Can anyone suggest a direction, is there a common issue or is it maybe my hardware etc....

Thank you

Kasper

![image](https://user-images.githubusercontent.com/14205873/214808377-2f03431f-4b5c-4c7b-8f44-24a201e3088d.png)

**Comment by fdufnews** [comment]
> When i press the board in lower right corner it re-connects and work again - no need to power cycle or anything

Looks like a bad solder joint.

Did you read Thonny's backend.log ?

**Comment by MariaA2005** [comment]
It's probably something not fully correctly connected (aka not correctly soldered). However, it could be an issue with the thonny version you have downloaded, make sure it's the official one and preferably the latest (uninstall it and get the correct one).

**Reply by rapzak** [reply]
I have noticed it is only when i write to the board - i tried to read several big files with no errors....
So i have think it may be when writing to flash, so either i think maybe bug in the write routines on stm32wb or the power supply drops during flash writing, due to more power consumption...
